packages/legacy/core/App/utils/proofBundle.ts

```diff
@@ -0,0 +1,135 @@
+import axios from 'axios'
+
+import { AnonCredsProofRequestTemplatePayload, ProofRequestTemplate, useProofRequestTemplates } from '../../verifier'
+import { useConfiguration } from '../contexts/configuration'
+
+const calculatePreviousYear = (yearOffset: number) => {
+  const pastDate = new Date()
+  pastDate.setFullYear(pastDate.getFullYear() + yearOffset)
+  return parseInt(pastDate.toISOString().split('T')[0].replace(/-/g, ''))
+}
+
+export const applyTemplateMarkers = (templates: any): any => {
+  if (!templates) return templates
+  const markerActions: { [key: string]: (param: string) => string } = {
+    now: () => Math.floor(new Date().getTime() / 1000).toString(),
+    currentDate: (offset: string) => calculatePreviousYear(parseInt(offset)).toString(),
+  }
+  let templateString = JSON.stringify(templates)
+  const markers = [...templateString.matchAll(/"@\{(\w+)(?:\((\S*)\))?\}"/gm)]
```

Comment:
Can we get a comment explaining that regex for ease of use if we come across it some months from now